cmd/ctr/commands/images/images.go

```diff
@@ -355,3 +357,77 @@ var removeCommand = cli.Command{
 		return exitErr
 	},
 }
+
+var pruneCommand = cli.Command{
+	Name:  "prune",
+	Usage: "remove unused images",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all", // TODO: add more filters
+			Usage: "remove all unused images, not just dangling ones (if all is not specified no images will be pruned)",
```

Comment:
I see nerdctl does the same, what's the limitation here that requires us to do all atm (if there is one)?

Reply:
I think it's because in `nertctl`/(now ctr as well) it always removes **unused images** (not referenced by containers, corresponds to`-a` flag) and `--filter` is not implemented. This means `nerdctl` always behaves like `docker image prune --all`.  

`docker image prune` (without `-a`) only deletes `dangling` images (no name/tag attached). Not sure if it (unnamed/untagged images) is supported in nerdctl/containerd...